dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ItemRestRepository.java

```diff
@@ -357,6 +360,28 @@ public Bundle addBundleToItem(Context context, Item item, BundleRest bundleRest)
         return bundle;
     }
 
+    /**
+     * Method to find the items for which the current user has editing rights.
+     *
+     * @param pageable Pagination information
+     * @return Page of Items (REST representation) for which the current user has editing rights
+     * @throws SearchServiceException
+     */
+    @SearchRestMethod(name = "findItemsWithEdit")
+    public Page<ItemRest> findItemsWithEdit(@Parameter(value = "query") String q, Pageable pageable)
```

Comment:
I think this also should have a `@PreAuthorize("hasAuthority('AUTHENTICATED')")`  You shouldn't be allowed to use this search endpoint until you are logged in.

We also should add a test to `ItemRestRepositoryIT` to prove that an anonymous user with not be able to used this endpoint and it'll result in a 401 exception

Reply:
Added the access restriction and a test.